### DISCUSSION TITLE
Converting cartesian tensor to irreps

### DISCUSSION THREAD

**Discussion by arunppsg** [e3nn/e3nn#350] [Q&A] [answered]
Hello,

I am new to equivariant networks and currently learning about the topic (my background in CS, so catching up with concepts like group theory, spherical harmonics etc). Sorry if this question is too naive.

I have a question about cartesian tensor: 
1. When we convert a cartesian tensor to irreps, say `x = CartesianTensor('ij=-ji'); t1 = x.from_cartesian(torch.ones(3, 3, 3))`, what is the math behind converting a cartesian tensor to irreps? Any reference to the topic on how it gets converted will be helpful.

2. I was experimenting with the library and I tried the following code snippet:
```
>>> x = CartesianTensor("ijk=-jik=-ikj")
>>> a = torch.ones(3, 3, 3)
>>> b = x.from_cartesian(a)
>>> c = x.to_cartesian(b)
>>> a - c
tensor([[[1., 1., 1.],
         [1., 1., 1.],
         [1., 1., 1.]],

        [[1., 1., 1.],
         [1., 1., 1.],
         [1., 1., 1.]],

        [[1., 1., 1.],
         [1., 1., 1.],
         [1., 1., 1.]]])
```
We are starting with `a`, a cartesian tensor, then converting it to irreps (`b`), then converting it back to cartesian tensr `c`. Now, `a` should be equal to `c` and I expected the output to be all zeros. But why is it all 1's?

3. Am I right in assuming that the irreps produced in the above example (`b`) are the Wigner-D matrices?

4. In the formula `ijk=-jik=-ikj`, does it mean by that the input tensors follow the symmetry?

**Comment by mariogeiger** [comment, marked as answer]
1. It's a change of basis between different representations of the group of rotation
2. Because your input has no antisymmetric part
3. Yes
4. Exactly, and that's why `a` is different from `c`

**Reply by arunppsg** [reply]
Hey, thanks for your response. Do you suggest any reference for understanding more about cartesian tensors and converting them to irreps - to get to know the math part?

**Reply by mariogeiger** [reply]
Maybe I can suggest 

Group Theory in a Nutshell for Physicists
by Anthony Zee

**Reply by arunppsg** [reply]
Thanks!
